### PR TITLE
Allow styling and customizing extended keyboard

### PR DESCRIPTION
**Pull request by skadyrov**
## Description:
- Make it possible to provide own layout for codeeditor (`codeEditorLayoutResourceId`). This allows the library user to configure/style the extended keyboard as they want
- Added `extendedKeyboardItemResourceId` field for custom extended key item layout
- Created `ExtendedKeyboardItemDivider`, and added `extendedKeyboardItemDivider` field for providing an instance of `ExtendedKeyboardItemDivider`

This PR addresses partially #6 as well.

## Screenshot:
screenshot_1541797431

### CODE REVIEW

File: library/src/main/res/layout/codeedditor_layout_default.xml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        android:paddingEnd="5dp"
+        android:layout_toEndOf="@id/lineNumbers"
+        android:layout_above="@id/extendedKeyboard"
+        android:background="@color/colorDocBackground"/>
+
+    <com.github.ahmadaghazadeh.editor.widget.ExtendedKeyboard
+        android:id="@+id/extendedKeyboard"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:elevation="5dp"
```
Comment (shockbytes) [inline library/src/main/res/layout/codeedditor_layout_default.xml]: Do we need elevation for the extended keyboard?

File: library/src/main/res/layout/extended_keyboard_list_item_default.xml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="utf-8"?>
+<TextView xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:id="@+id/tv_extended_key"
+    android:layout_width="wrap_content"
+    android:layout_height="wrap_content"
+    android:paddingStart="12dp"
```
Comment (shockbytes) [inline library/src/main/res/layout/extended_keyboard_list_item_default.xml]: Usually, lint warns you when you apply an asymmetric padding. Once again I would put this into the `dimens.xml` file.

File: .idea/vcs.xml [hunk 3]
```diff
@@ -2,5 +2,6 @@
 <project version="4">
```
Comment (shockbytes) [inline .idea/vcs.xml]: Same as above.

File: library/src/main/res/layout/codeedditor_layout_default.xml (hunk 1 above)
Comment (skadyrov) [reply library/src/main/res/layout/codeedditor_layout_default.xml]: yeah, for some reason the new lines drawn on `LineNumbersView` is shown over the top of the extended keyboard

File: library/src/main/res/layout/extended_keyboard_list_item_default.xml (hunk 2 above)
Comment (skadyrov) [reply library/src/main/res/layout/extended_keyboard_list_item_default.xml]: It warns if it is asymmetric horizontally or vertically. But it is ok to have different values for vertical and horizontal paddings

**Comment by skadyrov** [comment]
@shockbytes can you check again? let me know what you suggest for the package name

File: library/src/main/res/layout/extended_keyboard_list_item_default.xml (hunk 2 above)
Comment (shockbytes) [reply library/src/main/res/layout/extended_keyboard_list_item_default.xml]: Okay, fine :+1:

**Comment by shockbytes** [comment]
@skadyrov I think we can keep the package name. But you have to remove those `.idea`files from the commits. You already committed them, so you have to remove those files from the commit.

**Pull request merged**